### PR TITLE
DTO Validation/Service Implementation

### PR DESCRIPTION
**Pull request by kencas**
​
## Description
Implementation of the Post Data Validation and Implementation of the Layers of the code
​
## Related Issue (Link to linear ticket)

​
## Motivation and Context
Required to ensure that the Order Items are filled correctly
​
## How Has This Been Tested?
No test for now
​
## Screenshots (if appropriate - Postman, etc):
​
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
​
## Checklist:

- [ x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: prisma/schema.prisma [hunk 1]
```diff
@@ -175,4 +176,20 @@ model OrderReturn {
   user            User     @relation(fields: [userId], references: [id])
   product         Product  @relation(fields: [productId], references: [id])
   order           Order    @relation(fields: [orderId], references: [id])
+}
```
Comment (Megagig) [inline prisma/schema.prisma:179]: @kencas Changing this model resulted in a series of error in the project. Please kindly review this once more. 
	

- **Error validating: This line is not an enum value definition."
- This line is not an enum value definition
- Error validating: An enum must have at least one value.",**

File: prisma/schema.prisma [hunk 2]
```diff
@@ -175,4 +176,20 @@ model OrderReturn {
...
+  COMPLETED,
+  FAILED,
+}
+
+enum OrderStatus {
+  PENDING,
+  PROCESSING,
+  SHIPPED,
+  DELIVERED,
+  CANCELLED,
```
Comment (Megagig) [inline prisma/schema.prisma]: @kencas Please kindly check your enum definition here. There are matching errors on the terminal.

**Review by Megagig — CHANGES_REQUESTED** [review]
Hi @kencas 
Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

Highlights

1. No clear README description.
2. Altered the model schema resulting in series of errors
3. A lot of type errors on the terminal
4. No known route definition
5. Controller function not vividly defined.
6. Postman returns errors.

Required Changes ♻️
 Kindly remove the live demo section from your readme for now, you can add it when you have a deployed link 👍🏿
Check the comments under the review.

Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

You can also consider:

N/A
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification @Megagig 

Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.

**Comment by kencas** [comment]
> * No known route definition
> * Controller function not vividly defined.

No known route definition
Controller function not vividly defined. 

Explain

**Review by kencas — COMMENTED** [review]
Good to move on

**Comment by Megagig** [comment]
@kencas it is very unprofessional to merge a pull request without approval. Now see your merge has affected the database schema.

**Pull request merged**